osu.Game/Scoring/ScoreInfo.cs

```diff
@@ -66,44 +66,23 @@ public Mod[] Mods
                 if (mods != null)
                     return mods;
 
-                if (modsJson == null)
+                if (ModsJson == null)
                     return Array.Empty<Mod>();
 
-                return getModsFromRuleset(JsonConvert.DeserializeObject<DeserializedMod[]>(modsJson));
+                return getModsFromRuleset(JsonConvert.DeserializeObject<DeserializedMod[]>(ModsJson));
             }
             set
             {
-                modsJson = null;
                 mods = value;
+                ModsJson = mods == null ? null : JsonConvert.SerializeObject(mods.Select(m => new DeserializedMod { Acronym = m.Acronym }));
             }
         }
 
         private Mod[] getModsFromRuleset(DeserializedMod[] mods) => Ruleset.CreateInstance().GetAllMods().Where(mod => mods.Any(d => d.Acronym == mod.Acronym)).ToArray();
 
-        private string modsJson;
-
         [JsonIgnore]
         [Column("Mods")]
-        public string ModsJson
```

Comment:
what was wrong with this property?

Reply:
Since `Mods` is not mapped and `ModsJson` is mapped as `Mods` column, only changes to `ModsJson` can make efcore 5 detect changes in `Mods`. So here we should apply changes to `ModJson` to make sure changes are being tracked by efcore, and use `Mods` for json deserialization. Otherwise some tests will also fail.